server/impl/src/main/java/com/walmartlabs/concord/server/security/ldap/UserLdapGroup.java

```diff
@@ -0,0 +1,95 @@
+package com.walmartlabs.concord.server.security.ldap;
+
+import com.walmartlabs.concord.server.GenericOperationResult;
+import com.walmartlabs.concord.server.OperationResult;
+import com.walmartlabs.concord.server.sdk.ConcordApplicationException;
+import com.walmartlabs.concord.server.security.Roles;
+import com.walmartlabs.concord.server.user.UserInfoProvider;
+import com.walmartlabs.concord.server.user.UserManager;
+import com.walmartlabs.concord.server.user.UserType;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.Authorization;
+import org.apache.shiro.authz.UnauthorizedException;
+import org.sonatype.siesta.Resource;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Collections;
+import java.util.Set;
+import java.util.UUID;
+
+
+@Named
+@Singleton
+@Api(value = "UserLdapGroup", authorizations = {@Authorization("api_key"), @Authorization("session_key"), @Authorization("ldap")})
+@Path("/api/v1/userldapgroup")
+public class UserLdapGroup implements Resource {
+
+    private final UserManager userManager;
+    private final LdapUserInfoProvider ldapUserInfoProvider;
+    private final LdapGroupDao ldapGroupsDao;
+    private final LdapManager ldapManager;
+
+    @Inject
+    public UserLdapGroup(UserManager userManager, LdapUserInfoProvider ldapUserInfoProvider, LdapGroupDao ldapGroupsDao, LdapManager ldapManager) {
+        this.userManager = userManager;
+        this.ldapUserInfoProvider = ldapUserInfoProvider;
+        this.ldapGroupsDao = ldapGroupsDao;
+        this.ldapManager = ldapManager;
+    }
+
+    /**
+     * Sync Ldap groups for a ldap user
+     *
+     * @param username username
+     * @param domain   domain
+     * @return GenericOperationResult result
+     */
+    @POST
+    @ApiOperation("Sync ldap groups for a user")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/user/{username}/domain/{domain}/sync")
```

Comment:
Not sure why would a domain be a sub-resource of a user?
Perhaps it should be a
```
POST /api/v1/userldapgroup/sync
{
  "username": "...",
  "domain": "..."
}
```
request instead?

Reply:
updated